buildtools/BUILD.gn

```diff
@@ -128,6 +128,240 @@ source_set("gmock") {
   deps = [ "//gn:default_deps" ]
 }
 
+# Abseil-cpp configuration and source_set.
+# Required by protobuf v22+.
+config("abseil_config") {
+  visibility = _buildtools_visibility
+  include_dirs = [ "abseil-cpp" ]
+  cflags = []
+  if (is_clang) {
+    cflags += [
+      "-Wno-deprecated-builtins",
+      "-Wno-deprecated-pragma",
+      "-Wno-unknown-warning-option",
+    ]
+  }
+  if (is_gcc) {
+    cflags += [
+      "-Wno-stringop-overflow",
+      "-Wno-ignored-attributes",
+    ]
+  }
+}
+
+source_set("abseil_cpp") {
```

Comment:
can you add an assert_no_deps to make sure that nothing ends up depending on this (other than the protolite targets)
This should not end up in our binaries or in the amalgamated SDK